level1/recommendNewId.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class getNewId {
+    public static void main(String[] args){
+
+        Scanner scan = new Scanner(System.in);
+
+        String newId = scan.next();
+        String recommendId  = solution(newId);
+
+        System.out.println(recommendId);
+    }
+    
+    
+    public static String solution(String  new_id){
+
+        //1단계
+        String[] new_id_arr = new_id.toLowerCase().split("");
+        String CAPITAL = "[a-z\\d\\_\\-\\.]";
+        //2단계
+        for (int i = 0; i< new_id_arr.length;i++){
+            if (!Pattern.matches(CAPITAL,new_id_arr[i])){
+                new_id_arr[i] = "";
+            }
+        }
+        new_id = Arrays.stream(new_id_arr).collect(Collectors.joining());
+
+        //3단계
+        boolean flag = true;
+
+        while(flag){
+
+            if(new_id.indexOf("..")>-1) {
+                new_id = new_id.replace("..",".");
+                flag = true;
+            }else{
+                flag = false;
+            }
+
+        }
+        System.out.println(new_id);
+        //4단계
+        flag = true;
+        while(flag){
+
+            if(new_id.length() < 2){
+                if(".".equals(new_id)){
+                    new_id = "";
+                }
+                flag = false;
+            }else {
+                // 첫번째 . 제거
+                if (".".equals(new_id.substring(0, 1))) {
+                    new_id = new_id.substring(1, new_id.length());
+                } else {
+                    flag = false;
+                }
+
+            }
+
+
+            if(new_id.length() < 2){
+                if(".".equals(new_id)){
+                    new_id = "";
+                }
+                flag = false;
+            }else{
+
+                // 마지막 . 제거
+                if(".".equals(new_id.substring(new_id.length()-1,new_id.length()))){
+                    System.out.println("test");
+                    new_id = new_id.substring(0,new_id.length()-1);
+                    flag = true;
+                }else{
+                    flag = false;
+                }
+            }
+
+
+        }
+        //5단계
+        if("".equals(new_id.trim())){
+            new_id = "a";
+        }
+        //6단계
+        if (new_id.length() > 15){
+            new_id = new_id.substring(0,15);
+
+
+            flag = true;
+            while(flag){
+                if(".".equals(new_id.substring(new_id.length()-1,new_id.length()))) {
+                    new_id = new_id.substring(0,new_id.length()-1);
+                }else {
+                    flag = false;
+                }
+            }
+
+
+        }else if(new_id.length() < 3){
+            while(new_id.length() <3){
+                new_id += new_id.substring(new_id.length()-1, new_id.length());
+            }
```

Comment:
이부분은 약간 아쉬워요!!
혹시 문제의 조건이 80자 이상 200자 이하의 문자열을 받는 상황이었다면
반복문안에서 매번 substring으로 찾고있는데, 저 substring으로 찾는 문자는 몇번을 반복하던 변동이 없죠
```suggestion
String lastChar = new_id.substring(new_id.length()-1, new_id.length());
while(new_id.length() < 3) {
  new_id += lastChar;
}
```
이런식으로 사용한다면 반복문 안에서 불필요한 search가 사라질것 같아요!